code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -435,6 +437,95 @@
 	else
 		src << "<span class='danger'><B>You need to hold still!</span></B>"
 
+////Beam
+/obj/effect/ebeam/chain
+	name = "lightning chain"
+	layer = MOB_LAYER - 0.1
+
+/mob/living/simple_animal/hostile/guardian/beam
+	next_move_modifier = 0.5
+	melee_damage_lower = 7
+	melee_damage_upper = 7
+	attacktext = "shocks"
+	melee_damage_type = BURN
+	attack_sound = 'sound/machines/defib_zap.ogg'
+	damage_coeff = list(BRUTE = 0.7, BURN = 0.7, TOX = 0.7, CLONE = 0.7, STAMINA = 0, OXY = 0.7)
+	range = 8
+	playstyle_string = "As a lightning type, you will apply lightning chains to targets on attack and have a lightning chain to your summoner. Lightning chains will shock anyone near them."
+	magic_fluff_string = "..And draw the Tesla, a shocking, lethal source of power."
+	tech_fluff_string = "Boot sequence complete. Lightning modules active. Holoparasite swarm online."
+	var/datum/beam/summonerchain
+	var/datum/beam/enemychain
+	var/mob/living/enemy
+
+/mob/living/simple_animal/hostile/guardian/beam/AttackingTarget()
+	if(..())
+		if(isliving(target) && target != src && target != summoner && target != enemy)
+			if(enemychain)
+				qdel(enemychain)
+				enemychain = null
+			enemy = target
+			enemychain = Beam(enemy,"lightning[rand(1,12)]",'icons/effects/effects.dmi',70, 8,/obj/effect/ebeam/chain)
+
+/mob/living/simple_animal/hostile/guardian/beam/Life()
+	..()
+	if(summoner && summonerchain && !qdeleted(summonerchain))
+		chainshock(summonerchain)
+	else
+		summonerchain = null
+	if(enemy && enemychain && !qdeleted(enemychain))
+		chainshock(enemychain)
+	else
+		enemy = null
+		enemychain = null
+
+/mob/living/simple_animal/hostile/guardian/beam/Manifest()
+	if(..())
+		if(summoner)
+			summonerchain = Beam(summoner,"lightning[rand(1,12)]",'icons/effects/effects.dmi',INFINITY, INFINITY,/obj/effect/ebeam/chain)
+
+/mob/living/simple_animal/hostile/guardian/beam/Recall()
+	if(..())
+		if(summonerchain)
+			qdel(summonerchain)
+			summonerchain = null
+		if(enemychain)
+			qdel(enemychain)
+			enemychain = null
+			enemy = null
+
+/mob/living/simple_animal/hostile/guardian/beam/proc/chainshock(var/datum/beam/B)
+	var/list/turfs = list()
+	for(var/E in B.elements)
+		var/obj/effect/ebeam/chainpart = E
+		turfs |= get_turf(chainpart)
```

Comment:
get_turf_pixel() gives me a runtime;
![](http://puu.sh/nM806/98b13d61bf.png)
related code segment(other code is the same);
```
	for(var/E in B.elements)
		var/obj/effect/ebeam/chainpart = E
		var/turf/T = get_turf_pixel(chainpart)
		turfs |= T
		if(T != get_turf(B.origin) && T != get_turf(B.target))
			for(var/turf/TU in circlerange(T, 1))
				turfs |= TU
```